modules/cli/src/main/scala/scala/cli/commands/Run.scala

```diff
@@ -196,17 +208,20 @@ object Run extends ScalaCommand[RunOptions] {
         )
     }
 
-    if (retCode != 0)
-      if (exitOnError)
-        sys.exit(retCode)
-      else {
-        val red      = Console.RED
-        val lightRed = "\u001b[91m"
-        val reset    = Console.RESET
-        System.err.println(s"${red}Program exited with return code $lightRed$retCode$red.$reset")
-      }
+    process.onExit().thenApply { p1 =>
+      val retCode = p1.exitValue()
+      if (retCode != 0)
+        if (exitOnError)
+          sys.exit(retCode)
+        else {
+          val red      = Console.RED
+          val lightRed = "\u001b[91m"
+          val reset    = Console.RESET
+          System.err.println(s"${red}Program exited with return code $lightRed$retCode$red.$reset")
+        }
```

Comment:
I'm not sure we want to do that when we kill the process ourselves (this is going to print a weird `Program exited with return code 130`). We could keep the future that `thenApply` returns, and maybe call `cancel(false)` on it right before we try to kill it.

For that to be more convenient, maybe the call to `thenApply` should be done elsewhere, like when we call `maybeRun` (and only the call in watch mode should need fancy things).